lib/parse.js

```diff
@@ -52,16 +56,15 @@ function parse(input) {
           value = fastDecode(value) || value;
         }
 
-        if (result[key] === undefined) {
+        currentValue = result[key]
+        type = typeof currentValue;
+
+        if (type === "undefined") {
```

Comment:
String comparisons are always slower than === undefined or just checking for `.pop` to validate if it's array. I did a benchmark and just included the changes for this, and had a worse benchmark. Can you re-check your benchmarks?